### PR TITLE
[IPHLPAPI] Implement GetIpErrorString

### PR DESCRIPTION
**Pull request by tukitukybb**
## Purpose

Implement GetIpErrorString in iphlpapi_main.c.

NOTE: IP_STATUS_BASE+36..39 and +46..49 are unassigned gaps in the SDK header. Left out intentionally and falls through to ERROR_INVALID_PARAMETER.

JIRA issue: None

## Proposed changes
- Implement GetIpErrorString.
- Fix spacing in the signature.

## Testbot runs (Filled in by Devs)

- [ ] KVM x86:
- [ ] KVM x64:

### CODE REVIEW

**Comment by tukitukybb** [comment]
@MishaProductions, here's another one that might interest you.

**Comment by hiSandog** [comment]
For `GetIpErrorString`, the edge cases are probably as important as the known-code table. Tests or notes for unknown error codes, buffer length boundaries, and null/empty output buffers would make the implementation easier to validate against Windows behavior. That is especially useful for API compatibility work where callers may depend on exact fallback strings.

File: dll/win32/iphlpapi/iphlpapi_main.c [hunk 1]
```diff
@@ -2989,10 +2989,81 @@ DWORD WINAPI UnenableRouter(OVERLAPPED * pOverlapped, LPDWORD lpdwEnableCount)
...
-    return 0L;
+    unsigned int i;
+
+    static const struct
+    {
+        IP_STATUS Code;
+        const WCHAR *Text;
+    } IpStatusTable[] =
+    {
+        { IP_SUCCESS, L"The request succeeded." },
```
Comment (SigmaTel71) [inline dll/win32/iphlpapi/iphlpapi_main.c:3002]: These should be in resources instead of code I think. cc: @binarymaster

Comment (tukitukybb) [reply dll/win32/iphlpapi/iphlpapi_main.c:3002]: @binarymaster?

Comment (binarymaster) [reply dll/win32/iphlpapi/iphlpapi_main.c:3002]: ?

Comment (SigmaTel71) [reply dll/win32/iphlpapi/iphlpapi_main.c:3002]: @binarymaster, see my question above.

Comment (binarymaster) [reply dll/win32/iphlpapi/iphlpapi_main.c:3002]: No strong opinion on it, though localizability is better.

File: dll/win32/iphlpapi/iphlpapi_main.c [hunk 2]
```diff
@@ -2989,10 +2989,81 @@ DWORD WINAPI UnenableRouter(OVERLAPPED * pOverlapped, LPDWORD lpdwEnableCount)
 /*
  * @unimplemented
  */
-DWORD WINAPI GetIpErrorString(IP_STATUS ErrorCode,PWCHAR Buffer,PDWORD Size)
```
Comment (binarymaster) [inline dll/win32/iphlpapi/iphlpapi_main.c:2992]: Function annotation needs to be fixed, using SAL2.

Comment (tukitukybb) [reply dll/win32/iphlpapi/iphlpapi_main.c:2992]: Done! Thanks for reviewing :)

**Comment by tukitukybb** [comment]
Also, I've seen other PRs of implementing functions that use Doxygen documentation. Alongside replacing `@unimplemented` with `@implemented`, should I do that for this too?
